### PR TITLE
[Improve][Broker]Reduce PartitionedStats local REST call

### PR DESCRIPTION
**Pull request by AnonHxy**
### Motivation

* Reduce local REST calll for `org.apache.pulsar.broker.admin.impl.PersistentTopicsBase#internalGetPartitionedStats`

### Modifications

* Call local method if partition is owned by this broker,  or call REST api usng internal `AdminClient`

### Verifying this change

- [x] Make sure that the change passes the CI checks.

### Documentation

Check the box below or label this PR directly.

Need to update docs? 

- [ ] `doc-required` 
(Your PR needs to update docs and you will update later)
  
- [x] `doc-not-needed` 
(Please explain why)
  
- [ ] `doc` 
(Your PR contains doc changes)

- [ ] `doc-complete`
(Docs have been already added)

### CODE REVIEW

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java [hunk 1]
```diff
@@ -1342,16 +1343,35 @@ protected void internalGetPartitionedStats(AsyncResponse asyncResponse, boolean
                 return;
             }
             PartitionedTopicStatsImpl stats = new PartitionedTopicStatsImpl(partitionMetadata);
-            List<CompletableFuture<TopicStats>> topicStatsFutureList = Lists.newArrayList();
+            List<CompletableFuture<TopicStats>> topicStatsFutureList = new ArrayList<>(partitionMetadata.partitions);
+            AtomicBoolean exceptionFlag = new AtomicBoolean(false);
```
Comment (Technoboy-) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java:1347]: Maybe we can remove this line and check `exception` at line 1378

Comment (AnonHxy) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java:1347]: This check will break the loop if  throws `PulsarServerException`, that will avoid querying the left partitions stats. So I think it is useful @Technoboy-

Comment (Technoboy-) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java:1347]: `PulsarServerException` throw by `pulsar().getAdminClient()`. If this occurs, there will be a fast failed result.

Comment (AnonHxy) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java]: Nice catch. I get your point.  Have removed the `exceptionFlag`

**Comment by AnonHxy** [comment]
/pulsarbot CI - Build - MacOS / Unit-BROKER_FLAKY Tests (pull_request)

**Comment by AnonHxy** [comment]
@HQebupt   @codelipenghui @Jason918  PTAL ~

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java [hunk 2]
```diff
@@ -1342,17 +1342,31 @@ protected void internalGetPartitionedStats(AsyncResponse asyncResponse, boolean
...
+                            } else {
+                                CompletableFuture<TopicStats> restFuture = new CompletableFuture<>();
+                                try {
+                                    restFuture = pulsar().getAdminClient().topics().getStatsAsync(
+                                        partition.toString(), getPreciseBacklog, subscriptionBacklogSize,
+                                        getEarliestTimeInBacklog);
+                                } catch (PulsarServerException e) {
+                                    restFuture.completeExceptionally(e);
+                                }
+                                return restFuture;
```
Comment (Jason918) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java:1366]:
```suggestion
                                try {
                                    return pulsar().getAdminClient().topics().getStatsAsync(
                                        partition.toString(), getPreciseBacklog, subscriptionBacklogSize,
                                        getEarliestTimeInBacklog);
                                } catch (PulsarServerException e) {
                                  org.apache.pulsar.common.util.FutureUtil#failedFuture(e);
                                }
```
Reduce a CompletableFuture object if no exception.

Comment (AnonHxy) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java]: Nice suggestion. Have updated @Jason918

**Review by Jason918 — APPROVED** [review]

**Review by shibd — APPROVED** [review]

**Review by Technoboy- — APPROVED** [review]

**Pull request merged**